sklearn/ensemble/weight_boosting.py

```diff
@@ -581,7 +581,8 @@ def _boost_discrete(self, iboost, X, y, sample_weight, random_state):
         # Only boost the weights if I will fit again
         if not iboost == self.n_estimators - 1:
             # Only boost positive weights
-            sample_weight *= np.exp(estimator_weight * incorrect *
+            sample_weight = np.log(np.exp(sample_weight) +
```

Comment:
You have the exp and the log the wrong way around. You want to turn a product into a sum of logs.

Reply:
Ok,
    sample_weight = np.log(sample_weight * np.exp(estimator_weight*incorrect....))
This is fine I guess.

Reply:
No, you need to `exp(log(sample_weight) + est_weight * incorrect * ...)`